src/EventStateMachine.php

```diff
@@ -0,0 +1,352 @@
+<?php
+
+namespace RebelCode\State;
+
+use Dhii\Events\TransitionEventInterface;
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\State\ReadableStateMachineInterface;
+use Dhii\State\StateAwareTrait;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\State\Exception\CouldNotTransitionException;
+
+/**
+ * A readable, event-driven state machine.
+ *
+ * This implementation does not use an internal state graph. Instead, transition given to be applied will be
+ * used as the new state. As such, it is perfectly valid to transition to the same state, thus making this state
+ * machine implementation non-deterministic.
```

Comment:
Looks deterministic to me.

Reply:
Forgot to remove after changing the mechanics. Thanks for pointing out.